crates/ruff_python_formatter/src/statement/stmt_function_def.rs

```diff
@@ -60,21 +61,34 @@ impl FormatNodeRule<StmtFunctionDef> for FormatStmtFunctionDef {
 
         let format_inner = format_with(|f: &mut PyFormatter| {
             write!(f, [item.parameters.format()])?;
+
             if let Some(return_annotation) = item.returns.as_ref() {
                 write!(f, [space(), text("->"), space()])?;
-                if return_annotation.is_tuple_expr() {
-                    write!(
-                        f,
-                        [return_annotation.format().with_options(Parentheses::Never)]
-                    )?;
-                } else {
-                    write!(
-                        f,
-                        [optional_parentheses(
-                            &return_annotation.format().with_options(Parentheses::Never),
-                        )]
-                    )?;
-                }
+                write!(
+                    f,
+                    [maybe_parenthesize_expression(
+                        return_annotation,
+                        item,
+                        if empty_parameters(&item.parameters, f.context().source()) {
```

Comment:
This is meant to capture "Can the parameters break?". Like, semantically, it'd be nice if I could add a group around parameters and check if it breaks, but I don't think I can do that.

Reply:
You may be able to inspect the IR if you want. But I prefer AST based rules because they're easier understood

```rust
let mut recording = f.start_recording();

write!(recording, [item.parameters.format()])?;

let recorded = recording.stop();
// now inspect the document
recorded.iter().any(|element| ....)
```